### PR TITLE
completed task

### PR DESCRIPTION
**Pull request by BGBroth**

### CODE REVIEW

File: services/movie_session.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    if cinema_hall_id:
+        movie_session.cinema_hall_id = cinema_hall_id
+
+    movie_session.save()
+    return movie_session
+
+
+def delete_movie_session_by_id(session_id: int) -> None:
+    movie_session = MovieSession.objects.get(id=session_id)
```
Comment (Dimosphen1) [inline services/movie_session.py:53]: Use `get_movie_session_by_id` method here.

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested.

File: db/models.py [hunk 2]
```diff
@@ -14,3 +14,36 @@ class Actor(models.Model):
...
+
+
+class MovieSession(models.Model):
+    show_time = models.DateTimeField()
+    cinema_hall = models.ForeignKey(CinemaHall, on_delete=models.CASCADE)
+    movie = models.ForeignKey(Movie, on_delete=models.CASCADE)
+
+    def __str__(self) -> str:
+        return f"{self.movie.title} " \
+               f"{self.show_time.strftime('%Y-%m-%d %H:%M:%S')}"
```
[the remark this answers is not in the document]
Comment (Dimosphen1) [reply db/models.py:49]: This comment is not implemented, please place multi-line string inside of `()`.

File: services/cinema_hall.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+
+def get_cinema_halls() -> QuerySet:
+    return CinemaHall.objects.all()
+
+
+def create_cinema_hall(
+        hall_name: str,
+        hall_rows: int,
+        hall_seats_in_row: int) -> CinemaHall:
```
[the remark this answers is not in the document]
Comment (Dimosphen1) [reply services/cinema_hall.py:12]: This comment is not implemented, please place` ) -> CinemaHall:` on a new line

File: services/movie_session.py (hunk 1 above)
Comment (Dimosphen1) [reply services/movie_session.py:53]: This comment is not implemented.

File: services/movie_session.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        cinema_hall_id: int
+) -> None:
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie_id=movie_id,
+        cinema_hall_id=cinema_hall_id
+    )
+
+
+def get_movies_sessions(session_date: datetime = None) -> QuerySet:
```
[the remark this answers is not in the document]
Comment (Dimosphen1) [reply services/movie_session.py:18]: This comment is not implemented.

File: services/movie_session.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
+from db.models import MovieSession
+from django.db.models.query import QuerySet
+from datetime import datetime
```
[the remark this answers is not in the document]
Comment (Dimosphen1) [reply services/movie_session.py:3]: This comment is not implemented.

File: services/movie.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
+from django.db.models.query import QuerySet
+from db.models import Movie
+
+
+def get_movies(genres_ids: list = None, actors_ids: list = None) -> QuerySet:
```
[the remark this answers is not in the document]
Comment (Dimosphen1) [reply services/movie.py:5]: This comment is not implemented.

File: services/movie.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
+from django.db.models.query import QuerySet
+from db.models import Movie
```
[the remark this answers is not in the document]
Comment (Dimosphen1) [reply services/movie.py:2]: This comment is not implemented.

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
It seems that you requested a review without committing the requested changes. Please, request an additional review when all changes will be implemented.

**Comment by BGBroth** [comment]
I have encountered a Git error, now everything should work.

File: services/movie_session.py (hunk 1 above)
Comment (Dimosphen1) [reply services/movie_session.py:57]: This comment is still not implemented, please use `get_movie_session_by_id` here.

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Not all changes have been implemented.

**Review by Polyakiv-Andrey — CHANGES_REQUESTED** [review]
Tests not passed!

**Review by vasylhnatiuk — APPROVED** [review]

**Pull request closed without merging**